src/normalization/Normalizer.cpp

```diff
@@ -176,6 +176,15 @@ void Normalizer::normalizeMethod(Module& module, Method& method) const
     PROFILE_END(AddStartStopSegment);
 
     PROFILE_END(NormalizationPasses);
+
+    // add (runtime-configurable) loop over the whole kernel execution, allowing for skipping some of the syscall
```

Comment:
Moving the unrolling of work-groups into the normalization has the problem, that for all kernels there is a control-flow loop (which we actually don't care about), that we will have to handle!

Reply:
Have you already found a problem, if we make the loop?
I think, moving `unrolling work-groups` into first normalization is right way: Such program transformation should be earlier phase.
If not, we miss chances to optimize the loop.